packages/compass-assistant/src/components/assistant-chat.tsx

```diff
@@ -254,6 +271,53 @@ export const AssistantChat: React.FunctionComponent<AssistantChatProps> = ({
     }
   }, [hasNonGenuineConnections, chat, setMessages]);
 
+  const { activeConnections, activeWorkspace } = useAssistantGlobalState();
+
+  const activeConnection =
+    activeConnections.find((connInfo) => {
+      return (
+        hasConnectionId(activeWorkspace) &&
+        connInfo.id === activeWorkspace.connectionId
+      );
+    }) ?? null;
+
+  useEffect(() => {
```

Comment:
Wondering if this is the best place to be filling this in. I suppose it is OK since the concept of the "focused active connection" is a UI concept and how/where would we even determine that in other layers of the codebase.